client/components/app-promo/index.jsx

```diff
@@ -104,9 +79,27 @@ export default React.createClass( {
 						height="32"
 						alt="WordPress Desktop Icon"
 					/>
-					{ this.state.promo_item.message }
+					{ message }
 				</a>
 			</div>
 		);
 	}
 } );
+
+AppPromo.defaultProps = {
+	promoItem: getRandomPromo(),
+	translate: identity,
+	recordTracksEvent: noop,
+	saveDismissal: () => store.set( 'desktop_promo_disabled', true ),
+	getPromoLink
+};
+
+const mapDispatchToProps = ( dispatch ) => {
+	return {
+		recordTracksEvent: ( event, properties ) => {
+			dispatch( recordTracksEvent( event, properties ) );
+		}
+	};
+};
+
+export default connect( null, mapDispatchToProps )( localize( AppPromo ) ) ;
```

Comment:
As UI components become connected, they may fall under what's now being classified "blocks", which are typically those components bound to state.

See:
- https://github.com/Automattic/wp-calypso/blob/master/client/blocks/README.md
- p4TIVU-47o-p2